File: frontend/src/types/scoring.ts
// Scoring-related types based on the backend scoring schema

export interface UserScore {
  user_id: string;
  total_score: number;
  total_wins: number;
  total_losses: number;
  win_percentage: number;
}

export interface PickResult {
  id: string;
  user_id: string;
  game_id: string;
  player_id: string;
  status: "pending" | "win" | "loss" | "void";
  scored_at?: string;
  ftd_points: number;
  attd_points: number;
  total_points: number;
  is_manual_override: boolean;
  override_by_user_id?: string;
  override_at?: string;
  pick_submitted_at: string;
  created_at: string;
  updated_at?: string;
  // Related data
  player?: {
    id: string;
    name: string;
    team: string;
    position: string;
  };
  game?: {
    id: string;
    home_team: string;
    away_team: string;
    kickoff_time: string;
    week_number: number;
    status: string;
  };
  first_td_scorer?: {
    id: string;
    name: string;
    team: string;
    position: string;
  };
  all_td_scorers?: Array<{
    id: string;
    name: string;
    team: string;
    position: string;
  }>;
}

export interface GameResult {
  game_id: string;
  first_td_scorer_player_id?: string;
  all_td_scorer_player_ids: string[];
  scored_at?: string;
  is_manually_scored: boolean;
  picks_graded_count: number;
  // Related data
  first_td_scorer?: {
    id: string;
    name: string;
    team: string;
    position: string;
  };
  all_td_scorers?: Array<{
    id: string;
    name: string;
    team: string;
    position: string;
  }>;
  game?: {
    id: string;
    home_team: string;
    away_team: string;
    kickoff_time: string;
    week_number: number;
    status: string;
  };
}

File: backend/app/schemas/scoring.py
"""Scoring schemas for API responses"""

from pydantic import BaseModel, Field
from uuid import UUID
from datetime import datetime
from typing import Optional


class UserScore(BaseModel):
    """User's total score and statistics"""

    user_id: UUID
    total_score: int = Field(description="Total points from all winning picks")
    total_wins: int = Field(description="Number of winning picks")
    total_losses: int = Field(description="Number of losing picks")
    win_percentage: float = Field(description="Win percentage (0-100)")

    class Config:
        from_attributes = True


class PickResultResponse(BaseModel):
    """Detailed result for a single pick"""

    pick_id: UUID
    status: str = Field(description="win or loss")
    ftd_points: int = Field(description="Points from FTD prediction")
    attd_points: int = Field(description="Points from ATTD prediction")
    total_points: int = Field(description="Total points earned")
    actual_first_td_scorer: Optional[UUID] = Field(
        description="Player who scored first TD"
    )
    all_td_scorers: list[UUID] = Field(description="All players who scored TDs")
    scored_at: Optional[datetime] = Field(description="When pick was graded")

    class Config:
        from_attributes = True


class GameResultResponse(BaseModel):
    """Scoring results for a game"""

    game_id: UUID
    first_td_scorer: Optional[UUID] = Field(description="First TD scorer")
    all_td_scorers: list[UUID] = Field(description="All TD scorers")
    scored_at: Optional[datetime] = Field(description="When game was scored")
    picks_graded: int = Field(description="Number of picks graded")
    is_manually_scored: bool = Field(description="Whether manually scored by admin")

    class Config:
        from_attributes = True


class ManualScoringRequest(BaseModel):
    """Request to manually score a game"""

    first_td_scorer: Optional[UUID] = Field(description="First TD scorer player ID")
    all_td_scorers: list[UUID] = Field(description="All TD scorer player IDs")


class OverridePickRequest(BaseModel):
    """Request to override a pick's score"""

    status: str = Field(description="win or loss")
    ftd_points: int = Field(ge=0, le=3, description="FTD points (0 or 3)")
    attd_points: int = Field(ge=0, le=1, description="ATTD points (0 or 1)")
